src/target/riscv/riscv.c

```diff
@@ -3151,6 +3194,81 @@ COMMAND_HANDLER(riscv_set_ebreaku)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_itrigger)
+{
+	if (CMD_ARGC < 1) {
+		LOG_ERROR("Command takes at least 1 parameter");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	struct target *target = get_current_target(CMD_CTX);
+	const int unique_id = -CSR_TDATA1_TYPE_ITRIGGER;
+
+	if (riscv_enumerate_triggers(target) != ERROR_OK)
+		return ERROR_FAIL;
+
+	if (!strcmp(CMD_ARGV[0], "set")) {
+		if (find_first_trigger_by_id(target, unique_id) >= 0) {
```

Comment:
Minor stylistic thing - Here I'd recommend to have the unique ID for itrigger as a global constant. 

```
// somewhere near the top of riscv.c:
#define ITRIGGER_UNIQUE_ID (-CSR_TDATA1_TYPE_ITRIGGER)

// ...

// here in COMMAND_HANDLER(riscv_itrigger):
find_first_trigger_by_id(target, ITRIGGER_UNIQUE_ID)  // etc.
```

I understand that the unique ID is not used anywhere else outside this function (yet?). But I believe it will be quicker to the reader to understand the code when the uppercase constant is there, contrary to `unique_id` (which looks like an ordinary variable when reading the code).

Reply:
Another possibility: Leave the code as is but rename the variable

```
const int ITRIGGER_UNIQUE_ID = -CSR_TDATA1_TYPE_ITRIGGER;
```